sympy/utilities/iterables.py

```diff
@@ -315,7 +319,35 @@ def multiset(seq: Sequence[T]) -> dict[T, int]:
 
 
 
-def ibin(n, bits=None, str=False):
+@overload
+def ibin(
+    n: SupportsIndex,
+    bits: SupportsIndex | None = None,
+    str: Literal[False] = False,
+) -> list[int]: ...
+@overload
+def ibin(
+    n: SupportsIndex,
+    bits: SupportsIndex | None = None,
+    str: Literal[True] = True,
+) -> str: ...
+@overload
+def ibin(
+    n: SupportsIndex,
+    bits: str,
+    str: Literal[False] = False,
+) -> Iterable[tuple[int, ...]]: ...
+@overload
+def ibin(
+    n: SupportsIndex,
+    bits: str,
+    str: Literal[True],
+) -> Iterable[str]: ...
+def ibin(
```

Comment:
After all the overloads above I think it would be good to add two blank lines here to separate the actual function signature from the overloads for readability.

Like:
```
...
@overload
def ibin(
    n: SupportsIndex,
    bits: str,
    str: Literal[False] = False,
) -> Iterable[tuple[int, ...]]: ...
@overload
def ibin(
    n: SupportsIndex,
    bits: str,
    str: Literal[True],
) -> Iterable[str]: ...


def ibin(
    n: SupportsIndex,
    bits: SupportsIndex | str | None = None,
    str: bool = False,
) -> list[int] | str | Iterable[tuple[int, ...]] | Iterable[str]: ...
```